core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -680,12 +680,16 @@ abstract class AbstractFetcherThread(name: String,
        */
       val offsetAndEpoch = leader.fetchEarliestOffset(topicPartition, currentLeaderEpoch)
       val leaderStartOffset = offsetAndEpoch.offset
-      warn(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to current " +
-        s"leader's start offset $leaderStartOffset")
       val offsetToFetch = Math.max(leaderStartOffset, replicaEndOffset)
       // Only truncate log when current leader's log start offset is greater than follower's log end offset.
-      if (leaderStartOffset > replicaEndOffset)
+      if (leaderStartOffset > replicaEndOffset) {
+        warn(s"Truncate fully and start at leader's start offset $leaderStartOffset for partition $topicPartition " +
+          s"because the local replica logEndOffset $replicaEndOffset is smaller than leader's start offset.")
```

Comment:
> What do you think about the following wording to be more aligned with the other log message?
> 
> ```
> Truncate fully and reset fetch offset for partition $topicPartition from $current to the current leader's start offset $new because the local replica's end offset is smaller than the current leader's start offsets.
> ```
> 

In contrast, the overall meaning is similar, I don't feel a huge difference.

Reply:
I was trying to make it looks like the other one.

Reply:
> I was trying to make it looks like the other one.

Yes, I see, and I have updated. @dajac